### DISCUSSION TITLE
scripting in wikki.js

### DISCUSSION THREAD

**Discussion by nihalgurjar47** [requarks/wiki#6499] [Help / Questions]
I am performing action of onClick in script of wikki.js but this and so many this type of javascript functionality is not working?I want to perform some function on clicking of the button but I am unable to implement the functionality

**Comment by Quezaquo** [comment]
Hi, I encountered some similar problems creating click detection and copying content to my clipboard.

I managed to get an event from `onclick()` via an `eventListener`. So after that you can put your function to execute.

Here is the script:
```html
<script type="text/javascript">
  window.onload = function() {
    let items = document.querySelectorAll('.my-class');
    
    for (let i = 0; i < items.length; i++) {
      items[i].addEventListener("click", function() {
        alert(`Item ${i`);
      })
    }
  }
</script>
```
So from what I have understood and seen, putting `onclick()` events directly on your HTML element does not work.
I hope this helps you and that it works with `eventListeners`!
